tsl/src/nodes/decompress_chunk/exec.c

```diff
@@ -61,38 +61,32 @@ decompress_chunk_state_create(CustomScan *cscan)
 	chunk_state->csstate.methods = &chunk_state->exec_methods;
 
 	Assert(IsA(cscan->custom_private, List));
-	Assert(list_length(cscan->custom_private) == 6);
-	List *settings = linitial(cscan->custom_private);
-	chunk_state->decompression_map = lsecond(cscan->custom_private);
-	chunk_state->is_segmentby_column = lthird(cscan->custom_private);
-	chunk_state->bulk_decompression_column = lfourth(cscan->custom_private);
-	chunk_state->aggregated_column_type = lfifth(cscan->custom_private);
-	chunk_state->sortinfo = lsixth(cscan->custom_private);
+	Assert(list_length(cscan->custom_private) == DCP_Count);
+	List *settings = list_nth(cscan->custom_private, DCP_Settings);
+	chunk_state->decompression_map = list_nth(cscan->custom_private, DCP_DecompressionMap);
+	chunk_state->is_segmentby_column = list_nth(cscan->custom_private, DCP_IsSegmentbyColumn);
+	chunk_state->bulk_decompression_column =
+		list_nth(cscan->custom_private, DCP_BulkDecompressionColumn);
+	chunk_state->sortinfo = list_nth(cscan->custom_private, DCP_SortInfo);
+
```

Comment:
It's becoming difficult to work with the lists, so I added enums with numbers of the particular settings.